branchLayout/impl/src/test/java/com/virtuslab/branchlayout/unit/BranchLayoutTestSuite.java

```diff
@@ -49,6 +49,45 @@ public void withBranchSlideOut_givenNonRootExistingBranch_slidesOut() {
     assertEquals(children.get(1).getName(), childName1);
   }
 
+  @Test
+  public void shouldBeAbleToFindNextAndPreviousBranches() {
+    // given
+    String rootName = "root";
+    String parentName0 = "parent0";
+    String childName0 = "child0";
+    String childName1 = "child1";
+    String parentName1 = "parent1";
+    String childName2 = "child2";
+
+    /*-
+        root                          root
+          parent0
+            child0                      child0
+            child1                      child1
+          parent1
+            child2
+    */
+
+    List<IBranchLayoutEntry> childBranches0 = List.of(
+        new BranchLayoutEntry(childName0, /* customAnnotation */ null, List.empty()),
+        new BranchLayoutEntry(childName1, /* customAnnotation */ null, List.empty()));
+
+    val entry0 = new BranchLayoutEntry(parentName0, /* customAnnotation */ null, childBranches0);
+
+    List<IBranchLayoutEntry> childBranches1 = List.of(
+        new BranchLayoutEntry(childName2, /* customAnnotation */ null, List.empty()));
+
+    val entry1 = new BranchLayoutEntry(parentName1, /* customAnnotation */ null, childBranches1);
+    val rootEntry = new BranchLayoutEntry(rootName, /* customAnnotation */ null, List.of(entry0, entry1));
+    val branchLayout = new BranchLayout(List.of(rootEntry));
+
+    assertEquals(branchLayout.findNextEntry(childName1).getName(), parentName1);
+    assertEquals(branchLayout.findPreviousEntry(childName1).getName(), childName0);
+    assertEquals(branchLayout.findNextEntry(parentName0).getName(), childName0);
+    assertEquals(branchLayout.findPreviousEntry(childName0).getName(), parentName0);
+    assertEquals(branchLayout.findPreviousEntry(parentName1).getName(), childName1);
```

Comment:
I'd also add cases where the result of `findPreviousEntry` and `findNextEntry` are null.